engine/src/flutter/shell/platform/windows/host_window.cc

```diff
@@ -193,38 +195,72 @@ void SetChildContent(HWND content, HWND window) {
              client_rect.bottom - client_rect.top, true);
 }
 
+void AdjustAlongAxis(LONG dst_origin, LONG dst_size, LONG* origin, LONG* size) {
+  *size = std::min(dst_size, *size);
+  if (*origin < dst_origin)
+    *origin = dst_origin;
+  else
+    *origin = std::min(dst_origin + dst_size, *origin + *size) - *size;
+}
+
+RECT AdjustToFit(const RECT& container, const RECT& inside) {
```

Comment:
Nit: the name `container` and `inside` aren't immediately obvious. Also, I'd expect that `inside` would be before `container` since we're adjusting `container` to fit `inside` (similar to how `origin/`size` are after `dst_origin`/`dst_size`). What do you think of:

1. Renaming `inside` to `parent` and `container` to `child`
2. Making the `parent` argument before the `child` argument